client/lib/wpcom-undocumented/lib/site.js

```diff
@@ -166,6 +166,20 @@ UndocumentedSite.prototype.removeEmailFollower = function( followerId, callback
 	}, callback );
 };
 
+UndocumentedSite.prototype.setHeaderImage = function( imageData, callback ) {
+	this.wpcom.req.post( {
```

Comment:
I find that it's useful to return the post result, so we support both promise and callback style.
```
return this.wpcom.req.post( { //... } );
```